pilot/pkg/config/kube/gateway/testdata/route-binding.yaml

```diff
@@ -30,6 +30,8 @@ spec:
     allowedRoutes:
       namespaces:
         from: All
+      kinds:
+      - kind: HTTPRoute
```

Comment:
I don't get why this is rejected. It is HTTPRoute for HTTP... isn't that valid?

Reply:
Yes, this one is valid. I added this to test the case where the kind is explicitly set to a valid kind, because we had no such test, and therefore didn't catch a bug where supportedKinds wasn't being set correctly  (`s == kind` check in `generateSupportedKinds` didn't work).

> should we also bump the gateway-api dependency so we have test coverage for this?

I think there's still another test that will fail. I can either try to fix that in this PR, before we merge. Otherwise, I was planning to look at that one separately, and then bump the dependency when everything is working.